### PR TITLE
Fix project health for kotlin-dsl, kotlin-dsl-plugins

### PR DESCRIPTION
**Pull request by tresat**

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

File: platforms/core-configuration/kotlin-dsl-plugins/build.gradle.kts [hunk 1]
```diff
@@ -25,8 +31,13 @@ dependencies {
     compileOnly(libs.slf4jApi)
     compileOnly(libs.inject)
 
-    implementation(libs.futureKotlin("stdlib"))
     implementation(libs.futureKotlin("gradle-plugin"))
+    implementation(libs.futureKotlin("gradle-plugin-api"))
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl-plugins/build.gradle.kts:35]: 🤔 Why two dependency declarations for `gradle-plugin-api`?

File: platforms/core-configuration/kotlin-dsl/build.gradle.kts [hunk 2]
```diff
@@ -98,8 +105,11 @@ dependencies {
         because("ClassBytesRepositoryTest makes use of Groovydoc task.")
     }
     integTestImplementation(project(":internal-testing"))
+
     integTestImplementation(libs.mockitoKotlin)
 
+    integTestImplementation(project(":java-language-extensions"))
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl/build.gradle.kts:111]: ❓ Isn't this redundant given the existing `api` dependency?

File: platforms/core-configuration/kotlin-dsl/build.gradle.kts [hunk 3]
```diff
@@ -123,7 +133,7 @@ dependencies {
     testFixturesImplementation(libs.jacksonKotlin)
     testFixturesImplementation(libs.asm)
 
-    integTestDistributionRuntimeOnly(project(":distributions-basics"))
+    integTestDistributionRuntimeOnly(project(":distributions-full"))
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl/build.gradle.kts:136]: ❓ Why change from `basics` to `full` here?

Comment (tresat) [reply platforms/core-configuration/kotlin-dsl/build.gradle.kts:136]: All of this was just trying things to see if any of them fixed that error.

**Pull request closed without merging**